scalafmt-core/shared/src/main/scala/org/scalafmt/util/StyleMap.scala

```diff
@@ -36,17 +37,18 @@ class StyleMap(
     val styleBuilder = Array.newBuilder[ScalafmtConfig]
     startBuilder += 0
     styleBuilder += init
-    val disableBinPack = mutable.Set.empty[Token]
+    val disableBinPack = mutable.Map.empty[Token, BinPack.Unsafe]
     def warn(err: String)(implicit fileLine: FileLine): Unit = logger.elem(err)
     tokens.arr.foreach { tok =>
-      def changeStyle(style: ScalafmtConfig): Boolean = {
+      def changeStyle(style: ScalafmtConfig): Option[ScalafmtConfig] = {
         val changing = curr != style
-        if (changing) {
+        if (!changing) None
+        else {
           startBuilder += tok.left.start
           styleBuilder += style
-          curr = style
+          try Some(curr)
```

Comment:
Could we write this explicitely?
```scala
val changed = Some(curr)
curr = style
changed
```
since `Some(curr)` doesn't throw an exception this seems a weird usage of try and might be unclear to what's going on.